crates/astria-composer/src/executor/mod.rs

```diff
@@ -415,6 +423,41 @@ impl Executor {
 
         reason.map(|_| ())
     }
+
+    // check for mismatched configured chain_id and sequencer client chain_id
+    pub(crate) async fn ensure_configured_chain_id_matches_remote(&self) -> eyre::Result<()> {
```

Comment:
perhaps simplify with `get_sequencer_chain_id` and migrate the ensure to the `run_until_stopped` the `get_sequencer_chain_id` function could also not be a method taking self but a function taking just a sequencer client (see `get_latest_nonce` as example) ?

The function title is a reason this can fail, but it can also fail for errors in grabbing genesis, feels like the function has too many responsibilities to me, don't think the ensure necessarily needs to be it's own method if its "grab the data needed then ensure it's good"

Reply:
Done in b8d148f.